spec.bs

```diff
@@ -5662,6 +5603,113 @@ A <dfn>per signals url bid generator</dfn> is an [=ordered map=] whose [=map/key
+  1. If |parsedHeader| is null, return an empty [=list=].
+  1. Let |result| be a new [=list=] of [=origins=].
+  1. [=list/For each=] |entry| in |parsedHeader|:
+    1. If |entry| is not a [=string=], return an empty [=list=].
+    1. Let |parsedEntry| be the result of [=parsing an https origin=] on |entry|.
+    1. If |parsedEntry| is failure, return an empty [=list=].
+    1. [=list/Append=] |parsedEntry| to |result|.
+  1. Return |result|.
+</div>
+
+<div algorithm>
+To <dfn>fetch script</dfn> given a [=URL=] |url| and a [=script fetcher=] |fetcher|:
+  1. Let |request| be a new [=request=] with the following properties:
+    :   [=request/URL=]
+    ::  |url|
+    :   [=request/header list=]
+    ::  «`Accept`: `text/javascript`»
+    :   [=request/client=]
+    ::  `null`
+    :   [=request/mode=]
+    ::  "`no-cors`"
+    :   [=request/referrer=]
+    :: "`no-referrer`"
+    :   [=request/credentials mode=]
+    ::  "`omit`"
+    :   [=request/redirect mode=]
+    :: "`error`"
+
+    Issue: One of the side-effects of a `null` client for this subresource request is it neuters all
+    service worker interceptions, despite not having to set the service workers mode.
+
+    Issue: Stop using "`no-cors`" mode where possible
+    (<a href="https://github.com/WICG/turtledove/issues/667">WICG/turtledove#667</a>).
+  1. Let |fetchController| be the result of [=fetching=] |request| with [=fetch/useParallelQueue=]
+    set to true, and [=fetch/processResponse=] set to the following steps given
+    a [=response=] |response|:
+    1. If the result of [=validating fetching response headers=] given |response| is false:
+      1. [=fetch controller/Abort=] |fetchController|.
+      1. Set |fetcher|'s [=script fetcher/origins authorized for cross origin trusted signals=] to
+        an empty [=list=] of [=origins=].
+      1. Set |fetcher|'s [=script fetcher/script body=] to failure.
+    1. Set |fetcher|'s [=script fetcher/origins authorized for cross origin trusted signals=] to the
+      result of [=parsing allowed trusted scoring signals origins=] given |response|'s [=response/
+      header list=].
+    1. Let |bodyStream| be |response|’s [=response/body=]’s [=body/stream=].
+    1. Let |bodyReader| be result of [=ReadableStream/getting a reader=] from |bodyStream|.
+    1. Let |successSteps| be a set of steps that take a [=byte sequence=] |responseBody|, and
+      perform the following:
+      1. If [=validate fetching response mime and body=] with |response|, |responseBody| and
+        "`text/javascript`" returns false, set |fetcher|'s [=script fetcher/script body=] to
+        failure.
+      1. Otherwise, set set |fetcher|'s [=script fetcher/script body=] to |responseBody|.
+    1. Let |failureSteps| be a set of steps that take an [=exception=] <var ignore>e</var>, and
+      perform the following:
+      1. Set set |fetcher|'s [=script fetcher/script body=] to failure.

```

Comment:
I can't figure out if fetch calls processResponse on errors.  The fetch spec never actually mentions calling fetch/processResponse, just setting it / passing it as an argument, as far as I can see.

Reply:
I think fetch/processResponse gets copied to fetchParams/process response:

https://fetch.spec.whatwg.org/#ref-for-fetch-params-process-response:~:text=Let%20fetchParams%20be,is%20crossOriginIsolatedCapability.

And that will eventually get used in the function called here:
https://fetch.spec.whatwg.org/#ref-for-fetch-finale:~:text=Otherwise%2C%20run%20fetch%20response%20handover%20given%20fetchParams%20and%20response.

...and this does look like it's used for errors, though there is a good chance I am wrong since my ability to follow the fetch spec is poor.

...but I have a hard time following the fetch spec myself.

Reply:
I think you're right - I completely missed that they renamed "processResponse" to "process response" when passing it as an argument.  :(